### PR TITLE
refactor (shtml,cliFilter,sjs,README): Modifications of files

### PR DESCRIPTION
**Pull request by ghost**
1) cliFilter.js, sjs.js: Take a fixed varaible out of scope of
a function as a local-global one.

2) shtml.js: For the duplicated function, union in the lower case.

3) README.md, README.zh-CN: Fix formations and explainations.

4) context.js: Use symbols to make properties absolutely private.

### CODE REVIEW

File: lib/helper/shtml.js [hunk 1]
```diff
@@ -35,18 +25,21 @@ module.exports = function shtml(val) {
           return;
         }
 
-        value = String(value);
+        value = value.toString();
```
Comment (dead-horse) [inline lib/helper/shtml.js]: use `String(value)` to avoid `value.toString` not exist

Comment (ghost) [reply lib/helper/shtml.js]: Fixed

**Comment by ghost** [comment]
@dead-horse：Fixed. Thanks anyway!

File: lib/helper/shtml.js [hunk 2]
```diff
@@ -40,13 +30,16 @@ module.exports = function shtml(val) {
         if (value[0] === '/' || value[0] === '#') {
           return;
         }
-        const hostname = urlparse(value).hostname;
+
+        const hostname = urlparse(value).hostname.toLowerCase();
+        // If we don't have our hostname in the app.security.domainWhiteList,
+        // Just check for `shtmlConfig.domainWhiteList` and `ctx.whiteList`.
         if (!isSafeDomainDefault(hostname, domainWhiteList)) {
```
Comment (dead-horse) [inline lib/helper/shtml.js]: isSafeDomainDefault => isSafeDomain

**Comment by dead-horse** [comment]
others LGTM

File: lib/helper/shtml.js (hunk 2 above)
Comment (ghost) [reply lib/helper/shtml.js]: Fixed

**Comment by ghost** [comment]
@dead-horse：done :)

**Review by dead-horse — APPROVED** [review]

**Pull request merged**